tests/foreman/cli/test_errata.py

```diff
@@ -459,6 +466,149 @@ def test_negative_install_by_hc_name_without_org_info(self):
             )
         self.assertIn('Error: Could not find organization', context.exception.stderr)
 
+    @tier3
+    def test_positive_list_affected_chosts_by_erratum_restrict_flag(self):
+        """View a list of affected content hosts for an erratum filtered
+        with restrict flags. Applicability is calculated using the Library,
+        so that search must not limit to CV or LCE. Installability
+        is calculated using the attached CV, subject to the CV's own filtering,
+        so that search must limit to CV and LCE.
+
+        :id: 594acd48-892c-499e-b0cb-6506cea7cd64
+
+        :Setup: Errata synced on satellite server.
+
+        :Steps:
+
+            1. erratum list --erratum-id=<erratum_id>
+               --organization-id=<org_id> --errata-restrict-installable=1
+
+            2. erratum list --erratum-id=<erratum_id>
+               --organization-id=<org_id> --errata-restrict-installable=0
+
+            3. erratum list --erratum-id=<erratum_id>
+               --organization-id=<org_id> --errata-restrict-applicable=1
+
+            4. erratum list --erratum-id=<erratum_id>
+               --organization-id=<org_id> --errata-restrict-applicable=0
+
+
+        :expectedresults: List of affected content hosts for an erratum is
+            displayed filtered with corresponding restrict flags.
+
+        :CaseAutomation: automated
+
+        """
+        # Create list of uninstallabe errata by removing FAKE_2_ERRATA_ID
+        UNINSTAllABLE = FAKE_9_YUM_ERRATUM[1:5]
+        # Check that only installable errata are present in the list
+        erratum_list = Erratum.list(
+            {
+                'errata-restrict-installable': 1,
+                'content-view-id': self.content_view['id'],
+                'lifecycle-environment-id': self.env['id'],
+                'organization-id': self.org['id'],
+                'per-page': 1000,
+            }
+        )
+        # Assert the expected errata is in the list
+        assert self.CUSTOM_ERRATA_ID in str(
+            erratum_list
+        ), "Errata not found in list of installable errata"
+        # Assert the uninstallable errata are not in the list
+        for i in range(len(UNINSTAllABLE)):
+            assert (UNINSTAllABLE[i]) not in str(erratum_list), "Unexpected errata found"
+        # Check list of errata is not affected by installable=0 restrict flag
+        erratum_list = Erratum.list(
+            {
+                'errata-restrict-installable': 0,
+                'content-view-id': self.content_view['id'],
+                'lifecycle-environment-id': self.env['id'],
+                'organization-id': self.org['id'],
+                'per-page': 1000,
+            }
+        )
+        for i in range(len(FAKE_9_YUM_ERRATUM)):
```

Comment:
`range(len())` is considered anti-pattern in Python. Just go over values of iterable.
Also, `Erratum.list()` returns Python list, doesn't it? Please don't throw away more convenient data structure by converting it to string.

I'd do something like below, after confirming that `Erratum.list()` returns list of dicts, and that "Title" is the value we want to assert on. (Another candidate would be "Errata ID" - I don't have environment set up to check what they are.)
```python
erratum_title_list = [erratum['Title'] for erratum in erratum_list]
for erratum in FAKE_9_YUM_ERRATUM:
    assert erratum in erratum_title_list
```

Same comment applies to other usages of this pattern below.

Reply:
will change those, using erratum_id_list
thank you

Reply:
and key is errata-id